python/setup_project/config_location_page.py

```diff
@@ -47,24 +49,52 @@ def os_key(element):
 
         # remove the widgets from the layout
         for (label, path, browse, _) in os_widgets:
-            self.layout().removeWidget(label)
-            self.layout().removeWidget(path)
-            self.layout().removeWidget(browse)
+            ui.storage_grid_layout.removeWidget(label)
+            ui.storage_grid_layout.removeWidget(path)
+            ui.storage_grid_layout.removeWidget(browse)
 
         # add them back in
-        offset = 2
+        offset = 0
         for (row, (label, path, browse, os_current)) in enumerate(os_widgets):
-            self.layout().addWidget(label, row+offset, 0, 1, 1)
-            self.layout().addWidget(path, row+offset, 2, 1, 1)
+            ui.storage_grid_layout.addWidget(label, row+offset, 0, 1, 1)
+            ui.storage_grid_layout.addWidget(path, row+offset, 2, 1, 1)
             if os_current:
                 # current os gets browse setup and we track the path widget
-                self.layout().addWidget(browse, row+offset, 3, 1, 1)
+                ui.storage_grid_layout.addWidget(browse, row+offset, 3, 1, 1)
                 browse.pressed.connect(self._on_browse_pressed)
                 self._path_widget = path
             else:
                 # hide the browse button on all other os'es
                 browse.hide()
 
+        # hide path fields when distributed config mode is active
+        self._update_path_input_state()
+        ui.select_centralized_config.clicked.connect(self._update_path_input_state)
+        ui.select_distributed_config.clicked.connect(self._update_path_input_state)
+
+        # gray out distributed config mode if uploaded_config field doesn't exist
+        sg_connection = sgtk.platform.current_engine().shotgun
+        field_data = sg_connection.schema_field_read("PipelineConfiguration")
+        if "uploaded_config" not in field_data:
+            # site does not have an uploaded configuration field
+            ui.select_distributed_config.setEnabled(False)
+            ui.select_distributed_config.setToolTip(
```

Comment:
if you don't have the field, it will grey out the radio button and give a tooltip with info.